controllers/proxy.go

```diff
@@ -1,25 +1,61 @@
 package controllers
 
 import (
-	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecr"
+	"github.com/aws/aws-sdk-go/service/ecr/ecriface"
 	"github.com/quintilesims/d.ims.io/controllers/proxy"
 	"github.com/zpatrick/fireball"
-	//"github.com/aws/aws-sdk-go/service/ecr"
-	"github.com/aws/aws-sdk-go/service/ecr/ecriface"
+	"github.com/zpatrick/go-cache"
+	"log"
+	"net/http"
+	"time"
 )
 
+// ecr tokens last for 12 hours: https://github.com/aws/aws-sdk-go/blob/master/service/ecr/api.go#L1022
+const TOKEN_EXPIRY = time.Hour * 12
+
 type ProxyController struct {
 	ecr   ecriface.ECRAPI
 	proxy proxy.Proxy
+	cache *cache.Cache
 }
 
 func NewProxyController(ecr ecriface.ECRAPI, p proxy.Proxy) *ProxyController {
 	return &ProxyController{
 		ecr:   ecr,
 		proxy: p,
+		cache: cache.New(),
 	}
 }
 
-func (r *ProxyController) DoProxy(c *fireball.Context) (fireball.Response, error) {
-	return nil, fmt.Errorf("proxy controller implemented")
+func (p *ProxyController) DoProxy(c *fireball.Context) (fireball.Response, error) {
+	token, err := p.getRegistryAuthToken()
+	if err != nil {
+		log.Printf("[ERROR] Failed to get token for reverse proxy: %v", err)
```

Comment:
[nit] "[ERROR] Failed to get Registry auth token:" seems more aligned with what would actually be happening here